contracts/staking/contracts/src/staking_pools/MixinStakingPoolRewards.sol

```diff
@@ -191,51 +134,130 @@ contract MixinStakingPoolRewards is
             denominator.safeDiv(MIN_TOKEN_VALUE)
         );
 
-        // store cumulative rewards
-        cumulativeRewardsByPoolPtr[epoch] = IStructs.Fraction({
-            numerator: numeratorNormalized,
-            denominator: denominatorNormalized
-        });
-        cumulativeRewardsByPoolLastStored[poolId] = epoch;
+        // store cumulative rewards and set most recent
+        _setCumulativeReward(
+            poolId,
+            epoch,
+            IStructs.Fraction({
+                numerator: numeratorNormalized,
```

Comment:
Yeah, you're right. There could definitely be tighter bounds here. 

This is my thinking.

The max fixed-point value for a token is `10^27` (1 bn tokens with 18 decimal places), which consumes `90` bits.

The max fixed-point value for a base-10 token multiplier is `10^50` because `10^50 * 10^27 = 10^77` which consumes `256` bits.

So, theoretically, we could store the numerator and denominator up to `10^50` without loss of precision. 

Notes:
1. This upper-bound might have to be _slightly less_ than `10^50` (depending on the impl) because we're doing cross-multiplication, so during the computation the numerator could theoretically reach `2(10^27 * 10^50)`.
2. We know the numerator is always ≤ the denominator. So we could normalize by `10^k`; where `k` is either `0 if x≤50` or `(x - 50) if x>50`; where `10^x` is the nearest power of 10 that is greater than the denominator and `x > 50`.
3. We could get even tighter bounds using base-2, but it may be more difficult to reason about. Tbd!